### PR TITLE
Add support for optional arrow click handler and pointer cursor (#20)

### PR DESCRIPTION
**Pull request by robinv85**
Came up with the minimal required changes to add support for an `onClick` handler function prop.
This will automatically add the `cursor: pointer` to the arrow stroke.
For my use case it doesn't need the same on the arrow marker but just for the stroke (because multiple arrows can end in the same point), so not sure if those should be configurable separately for other use cases.
I also passed by default the `markerId, from.id, to.id` besides the click event to the handler.
Ideally it should also be possible to style a specific arrow when clicking (to set it as "selected").

### CODE REVIEW

File: src/ArcherContainer.js [hunk 1]
```diff
@@ -254,7 +254,12 @@ export class ArcherContainer extends React.Component<Props, State> {
         parentCoordinates,
       );
       const markerId = this.getMarkerId(from, to);
-      const clickHandler = this.props.onClick;
+      const clickHandler = this.props.onClick
+        ? (evt) => {
+          if (this.props.onClick) {
+            this.props.onClick(evt, markerId, from.id, to.id);
```
Comment (robinvenneman) [inline src/ArcherContainer.js]: @pierpo could also pass an object like `arrowData { marker, from, to }` or similar ... and perhaps this same event should also apply to the marker

**Comment by pierpo** [comment]
@robinvenneman Can you do a stupid example using your feature? (it can be really really stupid, look at the other examples hahaha)
It will help me understand exactly what you want and helps me to check if anything went wrong.
(until I set up proper automated tests)

**Comment by pierpo** [comment]
@robinvenneman Can you provide me an example or a use case? I'd like to know exactly what's needed to help you with this feature 😊

**Comment by robinvenneman** [comment]
@pierpo yeah what I did is just delete the relation when the arrow is clicked, most simple solution for what I was working on but ideally it would be able to do more I guess. I will see next week to find some time to add this example in the demo.

**Comment by robinvenneman** [comment]
@pierpo I've added a simple demo with clickable arrows, currently only the line is handling those events. The labels seem to intercept the pointer events on the arrow unless set to 'none' so this feature may require some more fine-tuning.

**Comment by pierpo** [comment]
@robinvenneman I am so sorry, I completely forgot to work on this. Is this still a feature you need?

**Comment by robinvenneman** [comment]
@pierpo Hey, no problem I also kind of forgot about this :) I implemented this on a private fork for this project I was working on and haven't looked at it since, not even sure if it's still in use right now.

If you think this is a good idea to add to the library then I still want to help out though if you'd like.

**Comment by pierpo** [comment]
Haha well I don't know, I wonder if having clickable arrows will happen often :thinking: 
I would suggest to wait for other people to ask for the feature before integrating it officially :blush:

**Comment by robinvenneman** [comment]
Ok, good idea :thumbsup:

**Comment by hashwin** [comment]
@pierpo definitely have a need for this. The use case here would be in the context of a flow chart.
Allowing you to click on an arrow will allow you to add a new node in that particular position.
Ideally, I should be able to add a '+' button with a click handler that sits on the middle of an arrow.

**Comment by hashwin** [comment]
> @pierpo definitely have a need for this. The use case here would be in the context of a flow chart.
> Allowing you to click on an arrow will allow you to add a new node in that particular position.
> Ideally, I should be able to add a '+' button with a click handler that sits on the middle of an arrow.

Though, looks like this is already achievable using the 'label' option on a relation.

**Comment by pierpo** [comment]
> > @pierpo definitely have a need for this. The use case here would be in the context of a flow chart.
> > Allowing you to click on an arrow will allow you to add a new node in that particular position.
> > Ideally, I should be able to add a '+' button with a click handler that sits on the middle of an arrow.
> 
> Though, looks like this is already achievable using the 'label' option on a relation.

Yes indeed 😊 Glad that you found out a simple solution.

**Comment by pierpo** [comment]
Noted. I'll add this to my TODO!

I think we should start over, the PR is too old. The codebase has completely changed :smile:

**Comment by pierpo** [comment]
Closed in favor of https://github.com/pierpo/react-archer/pull/183 😊

**Comment by pierpo** [comment]
The PR was too old, so I started over. Thank you @robinv85 for your work, and sorry that it took me SO long to handle this 😴

**Pull request closed without merging**